### PR TITLE
Dev/broker interapp validation

### PR DESCRIPTION
**Pull request by janstaelensskyline**
Note, has code from the #50 PR as wel NuGet Consolidation Check.
This to be able to read the nuget versions used.

### CODE REVIEW

File: Protocol/Error Messages/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public static IValidationResult InvalidInterAppReplyLogic(IValidate test, IReada
                 Source = Source.Validator,
                 FixImpact = FixImpact.NonBreaking,
                 GroupDescription = "",
-                Description = String.Format("Invocation of method '{0}.{1}' is not compatible with '{2}'. QAction ID '{3}'.", methodClassType, incompatibleMethod, "DataMiner.Core.InterApp > v1.0.1.0", qactionId),
+                Description = String.Format("Invocation of method '{0}.{1}' is not compatible with '{2}'. QAction ID '{3}'.", methodClassType, incompatibleMethod, "DataMiner.Core.InterApp > v1.0.1.1", qactionId),
```
Comment (janstaelensskyline) [inline Protocol/Error Messages/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs:29]: 1.0.1.0 was correct. The first version released is 1.0.1.1  and there was a maximum length in the windows form app so I couldn't write >=1.0.1.1 hence the >1.0.1.0

File: Protocol/ErrorMessages.xml [hunk 2]
```diff
@@ -12094,7 +12094,7 @@
 									<InputParameters>
 										<InputParameter id="0">methodClassType</InputParameter>
 										<InputParameter id="1">incompatibleMethod</InputParameter>
-										<InputParameter id="2" value="DataMiner.Core.InterApp &gt; v1.0.1.0">incompatibleComponent</InputParameter>
+										<InputParameter id="2" value="DataMiner.Core.InterApp &gt; v1.0.1.1">incompatibleComponent</InputParameter>
```
Comment (janstaelensskyline) [inline Protocol/ErrorMessages.xml:12097]: same as above comment

File: Protocol/Tests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs [hunk 3]
```diff
@@ -130,21 +111,22 @@ private void CheckForSendMessage(CallingMethodClass callingMethod)
...
             {
                 // Someone passing along a different variable possibly defined earlier as the AgentId.
-                var symbolInfo = semanticModel.GetSymbolInfo(identifierName).Symbol;
+                var symbolInfo = RoslynHelper.GetSymbol(identifierName, semanticModel);
 
                 if (IsArgumentPropertyFromReturnAddress(symbolInfo))
                 {
-                    results.Add(Error.InvalidInterAppReplyLogic(test, qAction, qAction, "Message", "Send(ReturnAddress", qAction.Id.RawValue));
+                    results.Add(Error.InvalidInterAppReplyLogic(test, qAction, qAction, "Message", "Send(ReturnAddress", qAction.Id.RawValue)
+                                     .WithCSharp(callingMethod));
```
Comment (janstaelensskyline) [inline Protocol/Tests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs:119]: .WithCSharp  did not work when the compilation did not work.

File: Protocol/Tests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs [hunk 4]
```diff
@@ -179,37 +160,42 @@ private void CheckForSetParameter(CallingMethodClass callingMethod)
...
-                                return true;
-                            }
-                        }
-
-                        return false;
+                        return true;
                     }
                 }
+
+                return false;
```
Comment (janstaelensskyline) [inline Protocol/Tests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs:198]: Doesn't this only lets the foreach run once? If that is our goal then we should use .FirstOrDefault on the enumeration.
In the original code I was looping through all definitions to try and cover all bases, but I do also think there will always ever be a single one.

File: ProtocolTests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport/CSharpCoreInterAppBrokerSupport.cs [hunk 5]
```diff
@@ -39,8 +52,22 @@ public void QAction_CSharpCoreInterAppBrokerSupport_Valid()
         #region Invalid Checks
 
         [TestMethod]
-#if !NET6_0_OR_GREATER
-        [Ignore]
+        [Ignore("Isn't really relevant and causes other checks to fail")]
+        public void QAction_CSharpCoreInterAppBrokerSupport_InvalidInterAppReplyLogic_XmlBased()
```
Comment (janstaelensskyline) [inline ProtocolTests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport/CSharpCoreInterAppBrokerSupport.cs:56]: Why add this if it gets ignored?

File: ProtocolTests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport/CSharpCoreInterAppBrokerSupport.cs [hunk 6]
```diff
@@ -75,14 +102,14 @@ public void QAction_CSharpCoreInterAppBrokerSupport_InvalidInterAppReplyLogic()
...
-                        
+
             var expected = new ValidationResult
             {
                 Severity = Severity.Critical,
                 Certainty = Certainty.Certain,
                 FixImpact = FixImpact.NonBreaking,
                 GroupDescription = "",
-                Description = "Invocation of method 'methodClassType.incompatibleMethod' is not compatible with 'DataMiner.Core.InterApp > v1.0.1.0'. QAction ID 'qactionId'.",
+                Description = "Invocation of method 'methodClassType.incompatibleMethod' is not compatible with 'DataMiner.Core.InterApp > v1.0.1.1'. QAction ID 'qactionId'.",
```
Comment (janstaelensskyline) [inline ProtocolTests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport/CSharpCoreInterAppBrokerSupport.cs:112]: same remark as earlier

File: ProtocolTests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport/CSharpCoreInterAppBrokerSupport.cs (hunk 5 above)
Comment (MichielOda) [reply ProtocolTests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport/CSharpCoreInterAppBrokerSupport.cs:56]: Cause the management tool will generate the XMLs anyway and another unittest checks if the xml files are actually used in an unit test

File: Protocol/Tests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs (hunk 4 above)
Comment (MichielOda) [reply Protocol/Tests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs:198]: As discussed at the desk, this return was in the original code, but can be removed.

File: Protocol/Tests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs (hunk 3 above)
Comment (MichielOda) [reply Protocol/Tests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs:119]: WithCSharp is purely for the navigation in DIS. Not sure what you mean with 'did not work'

File: Protocol/Error Messages/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs (hunk 1 above)
Comment (MichielOda) [reply Protocol/Error Messages/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs:29]: I'll have a look at the max length.

**Review by MichielOda — DISMISSED** [review]

**Review by MichielOda — APPROVED** [review]

**Pull request merged**